### PR TITLE
Typescript assingnment

### PR DESCRIPTION
**Pull request by razarehan28**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
```
Comment (Rutvi25) [inline tsconfig.json]: remove unnecessary code

File: src/calc.ts [hunk 2]
```diff
@@ -0,0 +1,549 @@
+//selecting various elements
+let btns = document.querySelectorAll(".btn");
+let number = document.querySelectorAll(".number");
+const deg = <HTMLElement>document.querySelector("#degree");
+let functionButton = <HTMLElement>document.querySelector(".function-btn");
+const operator = document.querySelectorAll(".operator");
+const equal = document.querySelector("#equal");
+let trigoButton = <HTMLElement>document.querySelector(".trigonometry-function");
+let second = <HTMLElement>document.querySelector("#secondBtn");
+let calScreen = <HTMLElement>document.querySelector(".calculator-screen");
```
Comment (Rutvi25) [inline src/calc.ts]: types can be given to the variables, i.e. HTMLInputElement or HTMLButtonElement etc.

File: src/calc.ts [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+function memoryClear() {
+  for (let i = 0; i <= memory.length; i++) {
+    memory.pop();
+  }
+  return memory;
+}
+function memoryRecall() {
+  return memory.pop().toString();
+}
+// function mod(num) {
```
Comment (Rutvi25) [inline src/calc.ts]: remove unnecessary code

**Review by Rutvi25 — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

File: calc.html [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        <div class="calculator-body">
+          <div class="btn-grid">
+            <!-- row-1 -->
+            <div class="row">
+              <div id="degree" class="btn btn-deg">DEG</div>
+              <div id="f-e" class="btn btn-fe">F-E</div>
+            </div>
+            <!-- row-2 -->
+            <div class="row">
+              <div id="memory-clear" class="btn">MC</div>
```
Comment (desaianiruddh) [inline calc.html]: id should be in camelCase convention

File: src/calc.ts [hunk 5]
```diff
@@ -0,0 +1,558 @@
...
+      firstChild.classList.add("hide");
+      secondChild.classList.remove("hide");
+    } else {
+      firstChild.classList.remove("hide");
+      secondChild.classList.add("hide");
+    }
+  }
+});
+function eventList(e: Event) {
+  let ch = e.target as HTMLElement;
```
Comment (desaianiruddh) [inline src/calc.ts]: needs to use meaningful name of variables

File: src/calc.ts [hunk 6]
```diff
@@ -0,0 +1,558 @@
...
+  } else {
+    memory.push(num);
+    let memory1 = memory.reduce((acc, index) => +acc + +index, 0);
+    memory.pop();
+    memory.push(memory1.toString());
+    calScreen.innerHTML = "";
+    return memory1;
+  }
+}
+function memoryMinus() {
```
Comment (desaianiruddh) [inline src/calc.ts]: add return type for function

File: src/calc.ts [hunk 7]
```diff
@@ -0,0 +1,558 @@
...
+  if (memory.length === 0) {
+    alert("Nothing is in the memeory");
+  } else {
+    memory.pop();
+  }
+  return memory;
+}
+function memoryStored() {
```
Comment (desaianiruddh) [inline src/calc.ts]: add return type for function

File: src/calc.ts [hunk 8]
```diff
@@ -0,0 +1,558 @@
...
+  } else {
+    calScreen.innerHTML = memory[i].toString();
+    i++;
+    if (i === memory.length) {
+      i = 0;
+    }
+  }
+  return memory;
+}
+function memoryClear() {
```
Comment (desaianiruddh) [inline src/calc.ts]: add return type

**Review by sheth-jay — APPROVED** [review]

**Pull request merged**